timetagger/app/front.py

```diff
@@ -3424,7 +3435,24 @@ def _draw_container(self, ctx, total_time, x1, y1, x2, y2):
                 target_hours = self._current_targets.get(period, 0)
                 if target_hours <= 0:
                     continue
-                factor = self._hours_in_range / divisor
+                free_hours = dt.get_free_hours_in_range(
+                    t1, t2, window.simplesettings.get("workdays")
+                )
+                cleaned_hours_in_range = self._hours_in_range - free_hours
+                if period == "day":
+                    factor = cleaned_hours_in_range / divisor
+                elif period == "week":
+                    factor = cleaned_hours_in_range / (
+                        divisor - window.simplesettings.get("workdays") * 24
+                    )
+                elif period == "month":
+                    factor = cleaned_hours_in_range / (
+                        divisor - window.simplesettings.get("workdays") * 24 * 4.33
+                    )
+                elif period == "year":
+                    factor = cleaned_hours_in_range / (
+                        divisor - window.simplesettings.get("workdays") * 24 * 52
+                    )
```

Comment:
This PR is looking really good! One final request to make this code easier to follow:
* Please rename `cleaned_hours_in_range` to `work_hours_in_range`.
* Please rewrite the for loop to being `for period in ["day", "week", "month", "year"]:` and replace `divisor` with the actual values.
* Put `free_days_per_week = window.simplesettings.get("workdays")` somewhere, and use that variable instead. Once I figured out that this was what the value means, it all fell into place :)

Sorry to be rather nitpicky but I'm sure someone else will make changes to this code *someday*, and any errors in this logic easily result in subtle bugs that are not obvious at first.

Reply:
I have also fixed the get_free_hours_in_range function. It now also works for scales smaller than 1 day